File: Katas/C#/7KYU/Maximum Product.cs
using System;

public class Kata
{
    public static int AdjacentElementsProduct(int[] array)
    {
        var maxProduct = array[0] * array[1];

        for (int i = 1, j = 2; j < array.Length; i++, j++)
        {
            maxProduct = Math.Max(array[i] * array[j], maxProduct);
        }
        return maxProduct;
    }
}

File: Katas/JavaScript/7KYU/Maximum Product.js
function adjacentElementsProduct(array) {

    let maxProduct = array[0] * array[1];

    for (let i = 1; i < array.length; i++) {

        product = array[i] * array[i + 1];

        if (product > maxProduct)
            maxProduct = product;

    }

    return maxProduct;

}

// A different Solution:

function adjacentElementsProduct(array) {

    let answer = [];
    for (i = 1; i < array.length; i++) {
        answer.push(array[i - 1] * array[i])
    }
    return Math.max.apply(null, answer);

}

// Or:

function adjacentElementsProduct(array) {

    let newArray = [];
    for (i = 0; i < array.length - 1; i++) {
        newArray.push(array[i] * array[i + 1])
    }
    return Math.max(...newArray)

}